spec/system/frontend/checkout_spec.rb

```diff
@@ -43,4 +43,37 @@
     click_button("Save and Continue")
     click_button("Place Order")
   end
+
+  it "completes as a guest" do
+    order = Spree::TestingSupport::OrderWalkthrough.up_to(:delivery, user: nil)
+    visit '/'
+
+    allow_any_instance_of( # rubocop:disable RSpec/AnyInstance
```

Comment:
For future reference, it looks like testing a guest order requires a lot of preparation. Probably it'd be nice to extract that into the testing support in core at some point.